src/varname.jl

```diff
@@ -26,29 +29,64 @@ julia> @varname x[:, 1][1+1]
 x[:,1][2]
 ```
 """
-struct VarName{sym, T<:Tuple}
+struct VarName{sym,T<:Lens}
     indexing::T
```

Comment:
Should we rename `indexing` to `lens` @phipsgabler ?

Reply:
I wanted to suggest that already, but expected someone to complain about incompatibility!  I'm all for it.